### ISSUE TITLE
azurerm_monitor_private_link_scoped_service makes undetected changes to private endpoint for azurerm_monitor_private_link_scope

### ISSUE THREAD

**Issue by fgarcia-cnb** [hashicorp/terraform-provider-azurerm#25936]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.0.7

### AzureRM Provider Version

3.98.0

### Affected Resource(s)/Data Source(s)

azurerm_monitor_private_link_scope, azurerm_monitor_private_link_scoped_service, azurerm_private_endpoint

### Terraform Configuration Files

```hcl
resource "azurerm_monitor_private_link_scope" "ampls" {
  name                = "fgtest-ampls"
  resource_group_name = "dandrianov20210823-rg"
}

resource "azurerm_private_endpoint" "endpoint" {
  name                = "fgtest-ampls-azuremonitor-pe"
  resource_group_name = "dandrianov20210823-rg"
  location = "westus2"
  subnet_id = var.subnet_id

  private_service_connection {
    is_manual_connection           = false
    name                           = "fgtest-ampls-azuremonitor-pc"
    subresource_names              = ["azuremonitor"]
    private_connection_resource_id = azurerm_monitor_private_link_scope.ampls.id
  }
}

resource "azurerm_monitor_private_link_scoped_service" "ampls_svc" {
  name                = "fgtest-amplsservice"
  resource_group_name = "dandrianov20210823-rg"
  scope_name          = azurerm_monitor_private_link_scope.ampls.name
  linked_resource_id  = "resourceID"
}
```

### Debug Output/Panic Output

```shell
None
```

### Expected Behaviour

all resources are created fine... but the 3rd resource (azurerm_monitor_private_link_scoped_service) causes changes to the "custom_dns_configs" attribute of the private endpoint. however, terraform does not bring these changes into state... instead it detects them in the "Objects have changed outside of Terraform" part of the run. 

### Actual Behaviour
How can we get the private endpoint to see the additional custom dns entries created by azurerm_monitor_private_link_scoped_service?

**Comment by neil-yechenwei** [comment]
Thanks for raising this issue. Maybe using "depends_on = [azurerm_monitor_private_link_scoped_service.ampls_svc]" in "azurerm_private_endpoint" is helpful.

**Comment by fgarcia-cnb** [comment]
nah, won't make a difference since the scope service doesnt present any new changes, and it doesnt detect the dns changes in the private endpoint

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against v3.x of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
